ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -14009,6 +14009,13 @@ void SemanticErrorChecker::checkBloom(IHqlExpression * bloom)
 
 void SemanticErrorChecker::checkJoin(IHqlExpression * join)
 {
+    IHqlExpression * atmost = join->queryAttribute(atmostAtom);
+    if (atmost && atmost->isConstant())
```

Comment:
I don't think you want the isConstant test - it may have other children (which may prevent it being constnat).  it is child(0) that you are checking, and the rest of the code will work the same.